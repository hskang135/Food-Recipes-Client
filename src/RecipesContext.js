import React from 'react';

const RecipesContext = React.createContext({
  foodname: [],
  recipes: [],
  addRecipe: () => {},
  deleteRecipe: () => {},
  updateRecipe: () => {},
  editRecipe: () => {}
});

export default RecipesContext;


