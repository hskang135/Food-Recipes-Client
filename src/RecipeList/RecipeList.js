import React, { Component } from 'react';
import RecipesContext from '../RecipesContext';
import RecipeItem from '../RecipeItem/RecipeItem';

export default class RecipeList extends Component {
  static defaultProps = {
    recipes: []
  }

  static contextType = RecipesContext;
  
  render() {
    const { recipes } = this.context;

    return(
      <section className='RecipeList'>
        <ul className='food_list'>
          {recipes.map(recipe => 
            <RecipeItem
              key={recipe.id}
              id = {recipe.id}
              {...recipe}
            />
          )}
        </ul>
      </section>
    )
  }

};





