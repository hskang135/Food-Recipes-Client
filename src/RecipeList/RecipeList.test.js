import React from 'react';
import toJson from 'enzyme-to-json';
import RecipeList from './RecipeList';
import renderer from 'react-test-renderer';
import RecipesContext from '../RecipesContext';

const context = {
  recipes: [
    {
      "id": "cbc787a-ffaf-11e8-8sp2",
      "foodname": "Test food name",
      "ingrdients": "Test ingredients",
      "description": "Test description"
    },
    {
      "id": "cbc787a-ffaf-11e8-8ef4",
      "foodname": "Test food name",
      "ingrdients": "Test ingredients",
      "description": "Test description"
    },
    {
      "id": "cbc787a-ffaf-11e8-8y7i",
      "foodname": "Test food name",
      "ingrdients": "Test ingredients",
      "description": "Test description"
    },
  ]
}

describe(`RecipeList component`, () => {
  it('renders UI expeacted', () => {
    const tree = renderer
      .create(<RecipesContext.Provider value={context}> 
                <RecipeList /> 
              </RecipesContext.Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  // enzyme doesn't yet support React.createContext
  it.skip('renders Recipes in ul for each in array', () => {
    const props = {
      match: {
        params: {
          id: 'THIS_RECIPE_ID'
        }
      }
    }

    const ul = (<RecipeList {...props} />, context)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})



