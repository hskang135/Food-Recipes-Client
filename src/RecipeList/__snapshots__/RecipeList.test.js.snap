// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeList component renders UI expeacted 1`] = `
<section
  className="RecipeList"
>
  <ul
    className="food_list"
  >
    <div
      className="recipeItem"
    >
      <div
        className="recipeItem_info"
      >
        <h3>
          Test food name
        </h3>
        <ul>
          <li>
            Ingredients: 
          </li>
          <li>
            Description: 
            Test description
          </li>
        </ul>
      </div>
      <div
        className="recipesButton"
      >
        <button
          className="delete"
          onClick={[Function]}
        >
          Delete Recipe
        </button>
      </div>
    </div>
    <div
      className="recipeItem"
    >
      <div
        className="recipeItem_info"
      >
        <h3>
          Test food name
        </h3>
        <ul>
          <li>
            Ingredients: 
          </li>
          <li>
            Description: 
            Test description
          </li>
        </ul>
      </div>
      <div
        className="recipesButton"
      >
        <button
          className="delete"
          onClick={[Function]}
        >
          Delete Recipe
        </button>
      </div>
    </div>
    <div
      className="recipeItem"
    >
      <div
        className="recipeItem_info"
      >
        <h3>
          Test food name
        </h3>
        <ul>
          <li>
            Ingredients: 
          </li>
          <li>
            Description: 
            Test description
          </li>
        </ul>
      </div>
      <div
        className="recipesButton"
      >
        <button
          className="delete"
          onClick={[Function]}
        >
          Delete Recipe
        </button>
      </div>
    </div>
  </ul>
</section>
`;
